Assets/Scripts/HeartSystem.cs

```diff
@@ -17,61 +17,70 @@ public class HeartSystem : MonoBehaviour
     private void Start()
     {
         player = gameObject.GetComponent<Player>();
+        Player.healthChangedEvent += OnPlayerHealthChanged;
```

Comment:
Seems like making the `healthChangedEvent` non-static would simplify this, no? You'd be able to use the `player` object you just got right above.